llama_index/vector_stores/postgres.py

```diff
@@ -0,0 +1,158 @@
+from typing import List, Any, Type, Optional
+
+from llama_index.data_structs.node import DocumentRelationship, Node
+from llama_index.vector_stores.types import (
+    VectorStore,
+    NodeWithEmbedding,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+
+def get_data_model(base: Type, index_name: str) -> Any:
+    """
+    This part create a dynamic sqlalchemy model with a new table
+    """
+    from pgvector.sqlalchemy import Vector
+    from sqlalchemy import Column
+    from sqlalchemy.dialects.postgresql import BIGINT, VARCHAR, JSON
+
+    class AbstractData(base):  # type: ignore
+        __abstract__ = True  # tShis line is necessary
+        id = Column(BIGINT, primary_key=True, autoincrement=True)
+        text = Column(VARCHAR, nullable=False)
+        extra_info = Column(JSON)
+        doc_id = Column(VARCHAR)
+        embedding = Column(Vector(1536))  # type: ignore
+
+    tablename = "data_%s" % index_name  # dynamic table name
+    class_name = "Data%s" % index_name  # dynamic class name
+    model = type(class_name, (AbstractData,), {"__tablename__": tablename})
+    return model
+
+
+class PGVectorStore(VectorStore):
```

Comment:
nit: could you add some simple unit tests to make sure this works?

Reply:
Added